i18n/en/docusaurus-plugin-content-docs/current/integration/asset-workflow.addressables.md

```diff
@@ -0,0 +1,364 @@
+  - If the asset download failed
+    - This includes cases where asset loading fails, since downloading occurs even when an asset is loaded without having downloaded the asset.
+- Reconnection processing details
+  - If the asset download failed
+    - Repeat the connection according to the retry strategy.
+
+Use [event notifications](#assets-addressables-event) if you want to run processing based on the status of retry processing.
+
+### Add processing triggered by asset download state {#assets-addressables-event}
+
+AssetProvider has the following event notifications.
+
+- OnDownloading
+  - Timing：Just before downloading
+  - Type：IObservable
+  - Parameters：None
+- OnDownloaded
+  - Timing：Immediately after download (including during processing)
+  - Type：IObservable
+  - Parameters：Download status
+    - Asset name, total bytes, downloaded bytes, percentages, etc.
+- OnConnectRetrying
+  - Timing：Just before retrying the connection
+  - Type：IObservable
+  - Parameters：Retry count
+    - The first time is `1` and the second time is `2`.
+    - `1` means the start of retry strategy running.
+- OnConnectRetried
+  - Timing：Immediately after connection retry is finished
+    - If the retry is canceled, it will not be notified.
+  - Type：IObservable
+  - Parameters：Retry result
+    - true: If the retry strategy is run and the retry is successful
+    - false: If the retry strategy is run and the retry is not successful finally
+
+### Encrypt and decrypt assets {#assets-addressables-crypto}
+
+:::caution
+Note that using this feature alone is weak protection for intellectual property.
+
+By using this feature, if someone obtains only encrypted assets, they will not be able to decrypt them.
+For example, even if the URL of the asset download destination is known, the asset cannot be decrypted, thus preventing misappropriation of the asset.
+
+However, if the encryption algorithm and the key used for decryption (implementation of ICryptoStreamFactory) are embedded in the application, the information necessary for decryption can be accessed and decrypted through reverse engineering.
+Therefore, for the purpose of intellectual property protection, it is necessary to prevent decryption from the application by reverse engineering in addition to this feature.
+:::
+
+Customize Addressables build scripts and ResourceProvider to encrypt and decrypt assets.
+The following classes are provided.
+
+- Build script to encrypt assets
+  - BuildScriptEncryptMode
+- ResourceProvider to decrypt assets
+  - CryptoAssetBundleProviderBase
+  - CryptoAssetBundleResource
+  - DownloadHandlerFileWithDecryption
+- Factory that creates the CryptoStream used to encrypt and decrypt the asset
+  - ICryptoStreamFactory
+
+These classes are related as follows.
+- Since the cryptographic algorithm to be used and the management of keys vary depending on the security level required by the application, an ICryptoStreamFactory is provided to provide for their processing.
```

Comment:
an ICryptoStreamFactory is provided to provide for their processing.
-> an ICryptoStreamFactory is provided to implement for their processing.

Reply:
Fixed!
https://github.com/extreal-dev/Extreal.Guide/pull/42/commits/46a5d9caaf19190a1628b4110e952f3ae7703baa